src/test_board.py

```diff
@@ -1,16 +1,114 @@
 import unittest
-from board import construct_board, print_board
+from board import construct_board, print_board, is_board_full, get_position_if_valid, add_marble_to_board
 
 class BoardTest (unittest.TestCase):
     def test_if_construct_board_return_correctly_initialized_array(self):
-
         board = construct_board()
-
         print_board(board)
 
         expected_board  = [[0] * 6] * 6
+        
         self.assertListEqual(board, expected_board)
     
+    def test_is_board_not_full_should_return_false(self):
+        board = [[1]*6 for _ in range(6)]
+        board[0][0] = 0
+        print_board(board)
+
+        result = is_board_full(board)
+        
+        self.assertFalse(result)
+    
+    def test_is_board_full_should_return_true(self):
+        board = [[1]*6 for _ in range(6)]
+        print_board(board)
+
+        result = is_board_full(board)
+        
+        self.assertTrue(result)
+    
+    def test_get_position_if_valid_should_return_array_position(self):
```

Comment:
A good practice is to have 1 assertion per test. I suggest you use test providers (I don't know how they are called in Python, but it's a set of data you run your test on)